### ISSUE TITLE
Crash puma 504 Gateway

### ISSUE THREAD

**Issue by aimak** [mailjet/mailjet-gem#162]
Hello,

Here's a stacktrace I'm having with the mailjet-gem. It makes my whole server crashing, which is not something our customers appreciate :) Is there any way to protect my code against this ?

```
 (JSON::ParserError)/versions/2.3.1/lib/ruby/2.3.0/json/common.rb:156:in `parse': 822: unexpected token at '<html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx</center>
</body>
</html>
'
	from /home/webuser/.rbenv/versions/2.3.1/lib/ruby/2.3.0/json/common.rb:156:in `parse'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/json/decoding.rb:21:in `decode'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/api_error.rb:14:in `initialize'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/connection.rb:88:in `new'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/connection.rb:88:in `handle_exception'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/connection.rb:75:in `rescue in handle_api_call'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/connection.rb:62:in `handle_api_call'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/connection.rb:48:in `post'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/resource.rb:247:in `save'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/resource.rb:270:in `save!'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/resource.rb:108:in `block in create'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/resource.rb:107:in `tap'
	from /home/webuser/web/webapp/shared/bundle/ruby/2.3.0/gems/mailjet-1.5.4/lib/mailjet/resource.rb:107:in `create'
```

This is my `Gemfile.lock` : 
```
    mailjet (1.5.4)
      activesupport (>= 3.1.0)
      rack (>= 1.4.0)
      rest-client
```

**Comment by lucyjosef** [comment]
Had the same problem using the mailjet gem, we finally use Net::HTTP to make a straight call to the api (this gem hasn't been updated since 2017, so I guess it's preferable to use the api itself).

**Comment by xab3r** [comment]
Fixed in v1.6.0
